src/AspNet.Security.OAuth.EVEOnline/EVEOnlineAuthenticationHandler.cs

```diff
@@ -25,31 +32,94 @@ public partial class EVEOnlineAuthenticationHandler : OAuthHandler<EVEOnlineAuth
-        request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+        string? accessToken = tokens.AccessToken;
 
-        using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
-        if (!response.IsSuccessStatusCode)
+        if (string.IsNullOrWhiteSpace(accessToken))
         {
-            await Log.UserProfileErrorAsync(Logger, response, Context.RequestAborted);
-            throw new HttpRequestException("An error occurred while retrieving the user profile.");
+            throw new InvalidOperationException("No Access token was returned in the OAuth token.");
         }
 
-        using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+        var tokenClaims = ExtractClaimsFromToken(accessToken);
+
+        foreach (var claim in tokenClaims)
+        {
+            identity.AddClaim(claim);
+        }
 
         var principal = new ClaimsPrincipal(identity);
-        var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+        var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, tokens.Response!.RootElement);
         context.RunClaimActions();
 
         await Events.CreatingTicket(context);
         return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
     }
 
+    /// <summary>
+    /// Extracts the claims from the token received from the token endpoint.
+    /// </summary>
+    /// <param name="token">The token to extract the claims from.</param>
+    /// <returns>
+    /// An <see cref="IEnumerable{Claim}"/> containing the claims extracted from the token.
+    /// </returns>
+    protected virtual IEnumerable<Claim> ExtractClaimsFromToken([NotNull] string token)
+    {
+        try
+        {
+            var securityToken = Options.SecurityTokenHandler.ReadJsonWebToken(token);
+
+            var nameClaim = ExtractClaim(securityToken, "name");
+            var expClaim = ExtractClaim(securityToken, "exp");
+
+            var claims = new List<Claim>(securityToken.Claims);
+
+            claims.Add(new Claim(ClaimTypes.NameIdentifier, securityToken.Subject.Replace("CHARACTER:EVE:", string.Empty, StringComparison.OrdinalIgnoreCase), ClaimValueTypes.String, ClaimsIssuer));
+            claims.Add(new Claim(ClaimTypes.Name, nameClaim.Value, ClaimValueTypes.String, ClaimsIssuer));
+            claims.Add(new Claim(ClaimTypes.Expiration, UnixTimeStampToDateTime(expClaim.Value), ClaimValueTypes.DateTime, ClaimsIssuer));
+
+            var scopes = claims.Where(x => string.Equals(x.Type, "scp", StringComparison.OrdinalIgnoreCase)).ToList();
+
+            if (scopes.Any())
+            {
+                claims.RemoveAll(x => scopes.Contains(x));
```

Comment:
What's the rationale behind removing these `scp` claims? Why not just leave them as-is and just use them to create the `urn:eveonline:scopes` claim?

Reply:
I have no recollection on why I put that line of code in, have removed and still working via the sample app.
Will update with line removed.